cpp/scripts/check-name-extentions.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+import re
+import os
+import sys
+import subprocess
+import argparse
+
+
+
+DEFAULT_DIRS = ["cpp/bench",
+                "cpp/comms/mpi/include",
+                "cpp/comms/mpi/src",
+                "cpp/comms/std/include",
+                "cpp/comms/std/src",
+                "cpp/include",
+                "cpp/examples",
+                "cpp/src",
+                "cpp/src_prims",
+                "cpp/test"]
+
+IGNORE_FILES = ""
+Search_Regex = re.compile(r"(#include.*.cuh|__device__|__global__)")
+
+def parse_args():
+    argparser = argparse.ArgumentParser("Runs clang-format on a project")
+    argparser.add_argument("-regex", type=str,
+                           default=r"[.](h|hpp|cpp)$",
```

Comment:
Yes, this is a disadvantage of current scheme. 
It can only check for positive violations, i.e. if we have C++ source files with CUDA. This doesnt check if a source file without any CUDA code is wrongly named .cu or .cuh. I am assuming this wont occur that often, esp after seeing the codebase. What do you think @teju85 ?
Personally, I dont mind adding checks for ensuring __device__/__global__ or *.cuh includes exist in all cu / cuh files, but I am not very sure if it includes all cases possible to define CUDA kernels

Reply:
hmm.. that's a good point, especially the ability to use thrust/cub API's for calling into CUDA kernels! Let's park this thought, for now.